app/cdap/components/ConnectionList/Components/ConnectionTabs/index.tsx

```diff
@@ -90,46 +93,49 @@ export default function ConnectionsTabs({
             {tabsData.data.map((connectorType, connectorTypeIndex) => (
               <ConnectionTab
                 role="button"
+                data-testid={`connectionstabs-eachtab-${connectionColumnIndex}-${connectorTypeIndex}`}
                 onClick={() => {
-                  if (index > 1) {
+                  if (connectionColumnIndex > 1) {
                     if (connectorType.canBrowse) {
-                      handleChange(connectorType, index);
+                      handleChange(connectorType, connectionColumnIndex);
                     }
                   } else {
-                    handleChange(connectorType, index);
+                    handleChange(connectorType, connectionColumnIndex);
                   }
                 }}
                 label={
-                  index > 1 ? (
+                  connectionColumnIndex > 1 ? (
```

Comment:
these nested ternary operators are really confusing and I don't think you need them - you can have optional parameters being based into TabLabelCanBrowse so you only need to return two different components. You can also pull it out into a function and pass in connectorType and index to make it easier to read.

Reply:
This code will be refactored in the connection list feature.
We are not modifying this code here in this PR as it is not related to the snackbar feature.